cognee/modules/graph/utils/get_graph_from_model.py

```diff
@@ -12,87 +20,94 @@ def get_graph_from_model(data_point: DataPoint, include_root = True, added_nodes
+                field_value,
+                nodes,
+                edges,
+                added_nodes,
+                added_edges,
+            )
+
+        elif (
+            isinstance(field_value, list)
+            and len(field_value) > 0
+            and isinstance(field_value[0], DataPoint)
+        ):
             excluded_properties.add(field_name)
 
             for item in field_value:
-                property_nodes, property_edges = get_graph_from_model(item, True, added_nodes, added_edges)
-
-                for node in property_nodes:
-                    if str(node.id) not in added_nodes:
-                        nodes.append(node)
-                        added_nodes[str(node.id)] = True
-
-                for edge in property_edges:
-                    edge_key = str(edge[0]) + str(edge[1]) + edge[2]
-
-                    if str(edge_key) not in added_edges:
-                        edges.append(edge)
-                        added_edges[edge_key] = True
-
-                for property_node in get_own_properties(property_nodes, property_edges):
-                    edge_key = str(data_point.id) + str(property_node.id) + field_name
-
-                    if str(edge_key) not in added_edges:
-                        edges.append((data_point.id, property_node.id, field_name, {
-                            "source_node_id": data_point.id,
-                            "target_node_id": property_node.id,
-                            "relationship_name": field_name,
-                            "updated_at": datetime.now(timezone.utc).strftime("%Y-%m-%d %H:%M:%S"),
-                            "metadata": {
-                                "type": "list"
-                            },
-                        }))
-                        added_edges[edge_key] = True
-            continue
-
-        data_point_properties[field_name] = field_value
+                n_edges_before = len(edges)
+                nodes, edges, added_nodes, added_edges = add_nodes_and_edges(
+                    data_point, field_name, item, nodes, edges, added_nodes, added_edges
+                )
+                edges = edges[:n_edges_before] + [
+                    (*edge[:3], {**edge[3], "metadata": {"type": "list"}})
+                    for edge in edges[n_edges_before:]
+                ]
+        else:
+            data_point_properties[field_name] = field_value
 
     SimpleDataPointModel = copy_model(
         type(data_point),
-        include_fields = {
+        include_fields={
             "_metadata": (dict, data_point._metadata),
         },
-        exclude_fields = excluded_properties,
+        exclude_fields=excluded_properties,
     )
 
-    if include_root:
-        nodes.append(SimpleDataPointModel(**data_point_properties))
+    nodes.append(SimpleDataPointModel(**data_point_properties))
 
     return nodes, edges
 
 
+def add_nodes_and_edges(
+    data_point, field_name, field_value, nodes, edges, added_nodes, added_edges
```

Comment:
Do we actually need nodes and edges here? Could we hold that data in `added_nodes` and `added_edges`? Just thinking if we can make it simpler.

Reply:
There is also an implementation where we don't have separate `nodes` and `added_nodes`, and `edges` and `added_edges`, but I tried one and ran into difficulties. I expect the reason why they were separated in the first place is because it's not ideal to pass down a very 'fat' dictionary in recursive function calls, and the `added_` dicts are a lot thinner. My view is that either approach is fine, but currently we have this one and imo it's not worth the effort to change it at this point in time.

Reply:
Yeah, ok. Let's keep it this way then.